src/turbomind/models/llama/LlamaLinear.h

```diff
@@ -34,29 +35,69 @@ class LlamaLinear {
                  Type                       type      = kGemm,
                  int*                       lora_mask = nullptr)
     {
+        float           alpha     = 1.0f;
+        float           beta      = 0.0f;
+        GemmS4F16::Type gemm_type = type == kFusedSiluFfn ? GemmS4F16::kFusedSiluFfn : GemmS4F16::kGemm;
```

Comment:
kGemm, kFusedSiluFfn are already defined in this header file. Don't use GemmS4F16